interview-questions.md

```diff
@@ -4,49 +4,74 @@ Answer the following questions. First, without external resources. Challenge you
 
-  Your answer:
+  Your answer: You need to first make a new directory called components. Inside components you make a new file with the name of the component. Make sure to follow the same React basic boilerplate with the name of your file in the appropriate places. Be sure to export the file. And make sure to import it into the App.js file. Inside the new file make the changes needed to diplay "Hello World". You can use a variety of different ways to write out the "Hello Word" (h1, div, p) however be sure it is in the new files return function. Inside the app.js file put <NewFile /> inside of the App.js return function. Once that is done you should see the contents of your new file on the browser after dragging the file into it
+
 
-  Researched answer:
+  Researched answer: React lets you define components as classes or functions. Components defined as classes currently provide more features which are described in detail on this page. To define a React component class, you need to extend React.Component
+  The only method you must define in a React.Component subclass is called render(). All the other methods described on this page are optional.
 
+  These methods are called in the following order when an instance of a component is being created and inserted into the DOM:
+constructor()
+static getDerivedStateFromProps()
+render()
+componentDidMount()
 
 
 4. What is JSX?
 
-  Your answer:
+  Your answer: JSX is the javascript in REACT. It allows us to write HTML in javascript
+
+  Researched answer: JSX allows us to write HTML elements in JavaScript and place them in the DOM without any createElement()  and/or appendChild() methods.
+
+JSX converts HTML tags into react elements.
+
+It is called JSX, and it is a syntax extension to JavaScript. We recommend using it with React to describe what the UI should look like. JSX may remind you of a template language, but it comes with the full power of JavaScript.
+
+JSX produces React “elements”. We will explore rendering them to the DOM in the next section. Below, you can find the basics of JSX necessary to get you started.
 
-  Researched answer:
 
 
 
 5. What is state in React?
 
-  Your answer:
+  Your answer: State is an object inside of a React componenet. You can manipulate the state in order to control the behavior of a component
 
-  Researched answer:
+  Researched answer: React components has a built-in state object.
+
+The state object is where you store property values that belongs to the component.
+
+When the state object changes, the component re-renders.
 
 
 
 6. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
+  Your answer: Not sure. Hoist means raise maybe raising something in javascript
+
+  Researched answer: In JavaScript, a variable can be declared after it has been used.
+
+In other words; a variable can be used before it has been declared.
```

Comment:
It CAN be, should be, no. It causes a lot of issues and will have unexpected results. Be sure to avoid doing so if you dont have to.